### PR TITLE
Add page types for HTTP

### PR DESCRIPTION
**Pull request by wbamberg**
Part of https://github.com/openwebdocs/project/issues/91.

Applies page types as defined in https://github.com/mdn/content/issues/22761, except:

- `http-content-security-policy-directive` -> `http-csp-directive`
- `http-feature-policy-directive` -> `http-permissions-policy-directive`

### CODE REVIEW

File: files/en-us/web/http/cors/errors/corsnotsupportingcredentials/index.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 title: >-
   Reason: Credential is not supported if the CORS header
+page-type: http-cors-error
   'Access-Control-Allow-Origin' is '*'
```
Comment (wbamberg) [inline files/en-us/web/http/cors/errors/corsnotsupportingcredentials/index.md:5]:
```suggestion
  'Access-Control-Allow-Origin' is '*'
page-type: http-cors-error
```

File: files/en-us/web/http/cors/errors/corsnotsupportingcredentials/index.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 title: >-
   Reason: Credential is not supported if the CORS header
   'Access-Control-Allow-Origin' is '*'
+page-type: http-cors-error
 slug: Web/HTTP/CORS/Errors/CORSNotSupportingCredentials
```
Comment (wbamberg) [inline files/en-us/web/http/cors/errors/corsnotsupportingcredentials/index.md:6]:
```suggestion
slug: Web/HTTP/CORS/Errors/CORSNotSupportingCredentials
page-type: http-cors-error
```

File: files/en-us/web/http/csp/errors/cspviolation/index.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 title: >-
   Content Security Policy: The page's settings blocked the loading of a
   resource: xyz
+page-type: guide
 slug: Web/HTTP/CSP/Errors/CSPViolation
```
Comment (wbamberg) [inline files/en-us/web/http/csp/errors/cspviolation/index.md:6]:
```suggestion
slug: Web/HTTP/CSP/Errors/CSPViolation
page-type: guide
```

File: files/en-us/web/http/headers/user-agent/firefox/index.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Firefox user agent string reference
 slug: Web/HTTP/Headers/User-Agent/Firefox
```
[the remark this answers is not in the document]
Comment (wbamberg) [reply files/en-us/web/http/headers/user-agent/firefox/index.md:3]: (this question doesn't have to block this PR)

File: files/en-us/web/http/csp/errors/index.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ---
 title: CSP errors and warnings (Content Security Policy)
 slug: Web/HTTP/CSP/Errors
+page-type: guide
```
Comment (teoli2003) [inline files/en-us/web/http/csp/errors/index.md:4]: Shouldn't this one be a landing page?

**Review by teoli2003 — APPROVED** [review]
👍

File: files/en-us/web/http/csp/errors/index.md (hunk 5 above)
Comment (wbamberg) [reply files/en-us/web/http/csp/errors/index.md:4]: I don't think so: there is no "http-csp-error" category of page for this page to provide an index of.

(This is not a great page: I assume the error messages quoted vary from one browser to another, and it doesn't actually explain what you might be able to do about any of the errors listed. 🐰 🕳️ )

**Pull request merged**